Formula/glibc.rb

```diff
@@ -95,43 +88,191 @@ class Glibc < Formula
-  depends_on MakeRequirement => :build
+  depends_on "linux-headers@5.15" => :build
   depends_on SedRequirement => :build
   depends_on BrewedGlibcNotOlderRequirement
   depends_on :linux
   depends_on LinuxKernelRequirement
 
-  # GCC 4.7 or later is required.
-  fails_with gcc: "4.6"
+  MINIMUM_MAKE_VERSION = "4.0".freeze
+  MINIMUM_GCC_VERSION = "6.2".freeze
 
-  def install
-    # Fix Error: `loc1@GLIBC_2.2.5' can't be versioned to common symbol 'loc1'
-    # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=869717
-    # Fixed in glibc 2.24
-    inreplace "misc/regexp.c", /^(char \*loc[12s]);$/, "\\1 __attribute__ ((nocommon));"
+  resource "make" do
+    url "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
+    sha256 "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+  end
+
+  resource "python3" do
+    url "https://www.python.org/ftp/python/3.10.5/Python-3.10.5.tgz"
+    sha256 "18f57182a2de3b0be76dfc39fdcfd28156bb6dd23e5f08696f7492e9e3d0bf2d"
+  end
+
+  resource "m4" do
+    url "https://ftp.gnu.org/gnu/m4/m4-1.4.19.tar.xz"
+    sha256 "63aede5c6d33b6d9b13511cd0be2cac046f2e70fd0a07aa9573a04a82783af96"
+  end
+
+  resource "bison" do
+    url "https://ftp.gnu.org/gnu/bison/bison-3.8.2.tar.xz"
+    sha256 "9bba0214ccf7f1079c5d59210045227bcf619519840ebfa80cd3849cff5a5bf2"
+  end
 
+  resource "bzip2" do
+    url "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz"
+    sha256 "ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269"
+  end
+
+  resource "gcc" do
+    url "https://ftp.gnu.org/gnu/gcc/gcc-10.4.0/gcc-10.4.0.tar.xz"
+    sha256 "c9297d5bcd7cb43f3dfc2fed5389e948c9312fd962ef6a4ce455cff963ebe4f1"
+  end
+
+  resource "mpfr" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.6.tar.bz2"
+    sha256 "cf4f4b2d80abb79e820e78c8077b6725bbbb4e8f41896783c899087be0e94068"
+  end
+
+  resource "mpc" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz"
+    sha256 "617decc6ea09889fb08ede330917a00b16809b8db88c29c31bfbb49cbf88ecc3"
+  end
+
+  resource "gmp" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2"
+    sha256 "498449a994efeba527885c10405993427995d3f86b8768d8cdf8d9dd7c6b73e8"
+  end
+
+  def vendor_make?
+    if which("make").blank?
+      true
+    else
+      make_version = Utils.safe_popen_read("make", "--version").lines.first.split.last
+      Version.new(make_version) < MINIMUM_MAKE_VERSION
+    end
+  end
+
+  def install
     # Setting RPATH breaks glibc.
     %w[
       LDFLAGS LD_LIBRARY_PATH LD_RUN_PATH LIBRARY_PATH
       HOMEBREW_DYNAMIC_LINKER HOMEBREW_LIBRARY_PATHS HOMEBREW_RPATH_PATHS
     ].each { |x| ENV.delete x }
 
+    # TODO: make an audit exception or come up with a better way
+    # to use /usr/bin/gcc.
+    mkdir buildpath/"gcc/bin"
+    ln_sf "/usr/bin/gcc", buildpath/"gcc/bin"
+    ln_sf "/usr/bin/g++", buildpath/"gcc/bin"
+    ENV["CC"] = buildpath/"gcc/bin/gcc"
+    ENV["CXX"] = buildpath/"gcc/bin/g++"
```

Comment:
You could probably do this.
```suggestion
    ENV["CC"] = DevelopmentTools.locate("gcc")
    ENV["CXX"] = DevelopmentTools.locate("g++")
```

Reply:
That's an interesting solution.  If I'm not mistaken, we don't ever install an unversioned `gcc` in the GCC formulae.  So that means that this should never find brewed `gcc`, because it will always be something like `gcc-5` or `g++-5` etc.

Reply:
You can also teach `DevelopmentTools.locate` to check `HOMEBREW_GCC_PATH` to make it easy to support users who have a new enough GCC that isn't `/usr/bin/gcc`.

Reply:
I test my hypothesis and it was right, brewed GCC is indeed ignored even when it is installed.  So this solves the last TODO!

Reply:
> You can also teach `DevelopmentTools.locate` to check `HOMEBREW_GCC_PATH` to make it easy to support users who have a new enough GCC that isn't `/usr/bin/gcc`.

Sorry if I was unclear, but I don't want it to search for brewed GCC because the GCC spec will break `glibc`.  I'd rather we keep things simple here for now by only allowing the use of the host `gcc` if `/usr/bin/gcc` is new enough and vendoring in all other cases.  Introducing other compiler configurations into the picture will just complicate matters further in ways that are not easy to anticipate.

Reply:
`HOMEBREW_GCC_PATH` is meant to function like `HOMEBREW_CURL_PATH` -- it provides a way for the user to provide an alternative that doesn't live in `/usr/bin` in case the one that comes with their system is too old.